assets/src/edit-story/components/panels/layerStyle.js

```diff
@@ -49,10 +49,32 @@ function LayerStylePanel({ selectedElements, onSetProperties }) {
   useEffect(() => {
     setState({ opacity });
   }, [opacity]);
-  const handleSubmit = (evt) => {
+
+  const updateProperties = useCallback(() => {
     onSetProperties(state);
+  }, [onSetProperties, state]);
```

Comment:
In stead of solving this in each individual panel, why not change `onSetProperties` (and perhaps rename it to something much better) in a single central place? It can then take an extra property, e.g. `force=false`, that can force an instant update to be used for `onSubmit` rather than the regular `onChange`.

This whole updating properties could use a good refactor - the whole thing is heavily duplicated through all the panels. Feel free to uproot it completely and replace with something much simpler - should be doable.

Also, what about all the other panels? This PR only addresses 4 out of 13 panels. Are the others all working as they should?

Reply:
This PR was supposed to be only a quickfix proposal for #494 and #432 (as discussed in #494). Also, there is more to be fixed, not only the reported here (see @wassgha comment), so I will let this as it is for now and back to this as soon as priorities of Sprint 25 are done. Sounds good? And yes, it is fixing only panels  that I detected the problems of #494 and #432.

Reply:
Hm, I get what you're saying, but I still think that it's as easy to fix the general case in `onSetProperties` (converting it to a debouncer) rather than fixing individual panels.

This just makes everything more complex and increases tech debt, not decreasing it. If something's going to need a refactor in the future (and this is), it's not really helping anything but the very short term.

Always leave the codebase in a better state than you found it 😄.

Reply:
I don't think that we have a bad codebase on this proposal. I mean, you are proposing a general solution without considering that some panels are not presenting the issue of #494 and #432. What we have here is only a pattern to cover special cases that `onSetProperties` is not solving by itself. Anyway, I will try your suggestion later, since it is supposed that will not add side effects.